site/pages/create-consequence-stops.page.tsx

```diff
@@ -0,0 +1,318 @@
+import { NextPageContext } from "next";
+import Link from "next/link";
+import { parseCookies } from "nookies";
+import { ReactElement, SyntheticEvent, useState } from "react";
+import { SingleValue } from "react-select";
+import { z } from "zod";
+import ErrorSummary from "../components/ErrorSummary";
+import Radios from "../components/form/Radios";
+import SearchSelect from "../components/form/SearchSelect";
+import Select from "../components/form/Select";
+import Table from "../components/form/Table";
+import TextInput from "../components/form/TextInput";
+import TimeSelector from "../components/form/TimeSelector";
+import { BaseLayout } from "../components/layout/Layout";
+import {
+    CONSEQUENCE_TYPES,
+    COOKIES_CONSEQUENCE_INFO,
+    COOKIES_CONSEQUENCE_TYPE_INFO,
+    DISRUPTION_SEVERITIES,
+    VEHICLE_MODES,
+    COOKIES_CONSEQUENCE_STOPS_ERRORS,
+} from "../constants";
+import { CreateConsequenceProps, PageState } from "../interfaces";
+import { createConsequenceStopsSchema, stopsImpactedSchema } from "../schemas/create-consequence-stops.schema";
+import { typeOfConsequenceSchema } from "../schemas/type-of-consequence.schema";
+import { flattenZodErrors, getDisplayByValue, getPageStateFromCookies } from "../utils";
+import { getStateUpdater } from "../utils/formUtils";
+
+const title = "Create Consequence Stops";
+const description = "Create Consequence Stops page for the Create Transport Disruptions Service";
+
+export interface ConsequenceStopsPageInputs extends Partial<z.infer<typeof createConsequenceStopsSchema>> {}
+export interface Stop extends z.infer<typeof stopsImpactedSchema> {}
+
+const CreateConsequenceStops = ({
+    inputs,
+    previousConsequenceInformation,
+}: CreateConsequenceProps<ConsequenceStopsPageInputs>): ReactElement => {
+    const [pageState, setPageState] = useState<PageState<Partial<ConsequenceStopsPageInputs>>>(inputs);
+    const stateUpdater = getStateUpdater(setPageState, pageState);
+    const [selected, setSelected] = useState<SingleValue<Stop>>(null);
+
+    const getOptionLabel = (stop: Stop) => {
+        if (stop.commonName && stop.indicator && stop.atcoCode) {
+            return `${stop.commonName} ${stop.indicator} ${stop.atcoCode}`;
+        } else if (stop.commonName && stop.atcoCode) {
+            return `${stop.commonName} ${stop.atcoCode}`;
+        } else {
+            return "";
+        }
+    };
+
+    const handleChange = (value: SingleValue<Stop>) => {
+        if (
+            !pageState.inputs.stopsImpacted ||
+            pageState.inputs.stopsImpacted.filter((data) => data.id === value?.id).length === 0
```

Comment:
We can use a .some rather than a .filter here:
```ts
if (!pageState.inputs.stopsImpacted ||
            pageState.inputs.stopsImpacted.some((data) => data.id === value?.id))
```

Reply:
done